### ISSUE TITLE
Render area doesn't resize with mesa 18.1.6 on wayland

### ISSUE THREAD

**Issue by trimental** [jwilm/alacritty#1517]
The drawing area of the window does not resize with mesa 18.1.6 on wayland (arch linux, gnome 3.28.3), whereas the window decorations do (because they don't use egl). This might be an issue with mesa 18.1.6 but I've seen examples of mesa 18.1.6 resizing working on wayland. This issue is not fixed by https://github.com/jwilm/alacritty/pull/1403.

**Comment by chrisduerr** [comment]
Thanks for reporting the issue. I can't reproduce this issue on X11 and Alacritty itself did not change, so it's possible that this is a bug in mesa or the window manager.

Which compositor are you using and have you tried any other compositors?

**Comment by trimental** [comment]
Yeah I've tried both mutter and weston, this is probably more to do with the wayland/egl driver for mesa. The only reason I filled this issue for alacritty is because when I run the glutin examples the resizing works perfectly unlike in alacritty. 

I'm trying to hunt down what causes this difference, https://github.com/mesa3d/mesa/commit/a9fb331ea7d1a78936ea8d8385e44cfd66f835c1f is probably the commit that causes the issue
